### PR TITLE
Minor improvements to WooPay test coverage

### PR DESCRIPTION
**Pull request by malithsen**
Related to https://github.com/Automattic/woocommerce-payments/issues/9591

This PR introduces a couple of minor improvements to WooPay automated tests.

#### Changes proposed in this Pull Request

- Add a unit test to ensure that Save My Info component is only rendered once.
- Add e2e tests to cover WooPay enable/disable checkbox as a merchant.

#### Testing instructions

* Make sure all automated tests pass

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9671` or branch name `fix/improve-woopay-test-coverage` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/improve-woopay-test-coverage) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 14bf32c7cb43588f9f70e06ab01bcbacddeffd03
- Build time: 2024-11-13 20:28:49 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: tests/e2e-pw/specs/merchant/woopay-setup.spec.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { getMerchant, getShopper } from '../../utils/helpers';
+import { activateWooPay, deactivateWooPay } from '../../utils/merchant';
+
+test.describe( 'WooPay setup', () => {
+	let merchantPage: Page;
+	let shopperPage: Page;
+	let wasWooPayEnabled: boolean;
+
+	test.beforeAll( async ( { browser } ) => {
+		shopperPage = ( await getShopper( browser ) ).shopperPage;
```
Comment (hsingyuc) [inline tests/e2e-pw/specs/merchant/woopay-setup.spec.ts]: Is this one needed?

**Review by hsingyuc — COMMENTED** [review]
Good work, @malithsen! 👍 

Should we also make sure classic checkout also renders the form only once?
```
it( 'should render the save user form when checkbox is checked for classic checkout', () => {
	expect( screen.getAllByTestId( 'save-user-form' ) ).toHaveLength( 1 );
} );
```

File: tests/e2e-pw/specs/merchant/woopay-setup.spec.ts (hunk 1 above)
Comment (malithsen) [reply tests/e2e-pw/specs/merchant/woopay-setup.spec.ts]: Good catch. Removed in https://github.com/Automattic/woocommerce-payments/pull/9671/commits/9e8355e46eaab599ce063b9ded2b0edbd42fa541

**Comment by malithsen** [comment]
> Should we also make sure classic checkout also renders the form only once?

@hsingyuc We already check for it 

Classic checkout: https://github.com/Automattic/woocommerce-payments/blob/9e8355e46eaab599ce063b9ded2b0edbd42fa541/client/components/woopay/save-user/test/checkout-page-save-user.test.js#L167
Blocks checkout: https://github.com/Automattic/woocommerce-payments/blob/9e8355e46eaab599ce063b9ded2b0edbd42fa541/client/components/woopay/save-user/test/checkout-page-save-user.test.js#L243

**Review by hsingyuc — APPROVED** [review]
Thank you for the update!

**Pull request merged**